src/Sfa.Das.Sas.Infrastructure/CourseDirectory/CourseDirectoryClient.cs

```diff
@@ -50,6 +65,20 @@ public CourseDirectoryClient(IInfrastructureSettings settings, ICourseDirectoryP
             return selectedProviders;
         }
 
+        public Dictionary<string, bool> GetHeiProviders()
+        {
+            var records = _convertFromCsv.CsvTo<HeiProviderCsvRecord>(LoadHeiProvidersFromVsts());
+
+            var heiProviders = (from heiProviderCsvRecord in records where heiProviderCsvRecord.UkPrn != null && heiProviderCsvRecord.OrgType == "Higher Education Organisation" select heiProviderCsvRecord.UkPrn).Distinct().ToList();
+
+            return heiProviders.ToDictionary(record => record, record => true);
```

Comment:
This should just stay a list since you're not using the key and you don't actually care about key collisions

you can use [ICollection<T>.Contains](https://msdn.microsoft.com/en-us/library/bhkz42b3%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396) on the list to compare so the return type would be ICollection<string>